kt/godot-library/src/main/kotlin/godot/core/KtObject.kt

```diff
@@ -48,11 +48,13 @@ abstract class KtObject {
             //Native object already exists, so we know the id and ptr without going back to the other side.
             rawPtr = config.ptr
             id = config.id
-            //Singletons are never initialized from here.
-            if (config.needBind) {
```

Comment:
I am not fond of reintroducing the IsSingleton function after removing it (on top of adding a second layer of branching in the init of KtObject). Once we add the option for people to generate their own API based on the GDExtensions in their Godot projects, we won't be able to keep it with internal visibility.